src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImpl.java

```diff
@@ -191,31 +191,60 @@ public CaseEvent createCaseEvent(final CaseEvent caseEvent, final Case newCase,
       // should we create an ad hoc action?
       createAdHocAction(category, caseEvent);
 
-      // transition case group status
-      CaseGroup caseGroup = caseGroupRepo.findOne(targetCase.getCaseGroupFK());
-      try {
-        caseGroupService.transitionCaseGroupStatus(caseGroup, caseEvent.getCategory(), targetCase.getPartyId());
-      } catch (CTPException e) {
-        //The transition manager throws an exception if the event doesn't cause a transition, however there are lots of
-        // events which do not cause CaseGroupStatus transitions, (this is valid behaviour).
-        log.debug(e.getMessage());
-      }
+      transitionCaseGroupStatus(targetCase, caseEvent);
 
       // if this is a respondent enrolling event
-      if (caseEvent.getCategory().toString().equals("RESPONDENT_ENROLED")) {
+      if (caseEvent.getCategory().equals(CategoryDTO.CategoryName.RESPONDENT_ENROLED)) {
         // are there other case groups that need updating
         List<CaseGroup> caseGroups = caseGroupService.transitionOtherCaseGroups(targetCase);
         checkCaseState(category, caseGroups, caseEvent, newCase);
+
       } else {
+
         // should a new case be created?
         createNewCase(category, caseEvent, targetCase, newCase);
```

Comment:
Do we need to hit this function for `SUCCESSFUL_RESPONSE_UPLOAD` or `COMPLETED_BY_PHONE`?
If not we could make it 
```
} else if (caseEvent.getCategory().equals(CategoryDTO.CategoryName.SUCCESSFUL_RESPONSE_UPLOAD) ||
            caseEvent.getCategory().equals(CategoryDTO.CategoryName.COMPLETED_BY_PHONE)) {
  ...
} else {
  createNewCase(category, caseEvent, targetCase, newCase);
  effectTargetCaseStateTransition(category, targetCase);
}
```
Which would flatten the code out a bit

Reply:
Unfortunately it needs to run createNewCase first. Some unit tests go bang otherwise. I was having a look at how to simplify the whole class but not sure yet.

Reply:
Cool I think it's good as it is anyway 👍